docs/user-interaction-config-file-spec.rst

```diff
@@ -0,0 +1,171 @@
+for both Anaconda and a Live installation environment.
+
+
+Configuration file location
+---------------------------
+
+The user interaction configuration file is stored in: ``/etc/sysconfig/anaconda``
+
+
+General configuration file systax
+---------------------------------
+
+The configuration file is based on the INI file de-facto standard,
+eq.: key=value assignments and square bracket framed section headers.
+
+Example:
+
+::
+
+    top_level_key=some_value
+
+    [section_1]
+    key_a_in_section1=some_value
+    key_b_in_section1=some_value
+
+    [section_2]
+    key_a_in_section2=some_value
+
+Boolean values are marked with 1 for true and 0 for false.
+
+Example:
+
+::
+
+    true_key=1
+    false_key=0
+
+
+Toplevel namespace
+------------------
+
+The toplevel configuration file namespace can only contain the
+``post_install_tools_disabled`` key and section headers.
+
+The ``post_install_tools_disabled`` key corresponds to using ``firstboot --disable`` command in the
+installation kickstart, that requests the post installation setup tools to be skipped. If this key is present
+and set to true, then any post installation tools that parses the user interaction file should make sure
+it won't be started again and then terminate immediately.
+
+Section headers each correspond to a spoke screen shown in Anaconda. Hubs are not represented as section
+as there is nothing that can be directly set on a hub in Anaconda.
+All currently visible spokes in Anaconda will be enumerated as a section headers by Anaconda in the
+configuration file. Note that this can include spokes provided by addons, if any Anaconda addons that provide
+additional spokes are present during the installation run.
+
+
+Section naming
+--------------
+
+Section headers are named according to the Anaconda spoke class name. For example ``DatetimeSpoke``
+or ``KeyboardSpoke``.
+
+
+Section namespace
+-----------------
+
+Each section *must* contain the ``visited`` key with a value of either true if the user has visited
+the corresponding screen or false if not.
+
+Optional each section can contain one or more keys with the ``changed_`` prefix which track if the user
+has changed an option on the screen. If the option is changed by the user, the corresponding key is set
+to true. If the given option has not been changed by the user then the corresponding key can either be
+omitted or set to false.
+
+Example:
+
+::
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+In this example the user has visited the date & time spoke and has changed the timezone & time/date,
+but not the NTP settings. Note that the ``changed_ntp`` key could also be omitted as the user has not changed
+the NTP options.
+
+Another example:
+
+::
+
+    [KeyboardSpoke]
+    visited=0
+
+Here the user has not visited the keyboard spoke and thus could not have changed any options,
+so all ``changed_*`` keys (if any) have been omitted.
+
+
```

Comment:
They should also assume that if a spoke isn't mentioned in the configuration file that it hasn't been visited.